### PR TITLE
OPRUN-3436: UPSTREAM: <carry>: add openshift kustomize overlay

### PR DESCRIPTION
**Pull request by everettraven**
to enable TLS communication with catalogd. Configure the CA certs using the configmap injection method via service-ca-operator

### CODE REVIEW

File: openshift/kustomize/overlays/openshift/patches/manager_deployment_ca.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+- op: add
+  path: /spec/template/spec/volumes/-
+  value: {"name":"olmv1-certificate", "configMap":{"name":"operator-controller-openshift-ca", "optional": false, "items": [{"key": "service-ca.crt", "path": "olm-ca.crt"}]}}
+- op: add
+  path: /spec/template/spec/containers/0/volumeMounts/-
+  value: {"name":"olmv1-certificate", "readOnly": true, "mountPath":"/var/certs/olm-ca.crt", "subPath":"olm-ca.crt"}
+- op: add
+  path: /spec/template/spec/containers/0/args/-
+  value: "--ca-certs-dir=/var/certs"
```
Comment (tmshort) [inline openshift/kustomize/overlays/openshift/patches/manager_deployment_ca.yaml:9]: Well.. the directory argument seems to be useful here...

File: openshift/manifests/14-configmap-openshift-operator-controller-operator-controller-openshift-ca.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
```
Comment (tmshort) [inline openshift/manifests/14-configmap-openshift-operator-controller-operator-controller-openshift-ca.yml:1]: Not needed? (but this is generated?)

**Review by tmshort — APPROVED** [review]
/lgtm

Two (small) comments that need not be handled.

File: openshift/manifests/14-configmap-openshift-operator-controller-operator-controller-openshift-ca.yml (hunk 2 above)
Comment (everettraven) [reply openshift/manifests/14-configmap-openshift-operator-controller-operator-controller-openshift-ca.yml:1]: It was indeed generated

**Pull request merged**